crates/span/src/hygiene.rs

```diff
@@ -26,12 +26,44 @@ use crate::InternId;
 #[cfg(feature = "ra-salsa")]
 use ra_salsa::{InternId, InternValue};
 
-use crate::MacroCallId;
+use crate::{Edition, MacroCallId};
 
 /// Interned [`SyntaxContextData`].
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Copy, Eq)]
 pub struct SyntaxContextId(InternId);
 
+// For comparisons etc. we remove the edition, because sometimes we search for some `Name`
+// and we don't know which edition it came from.
+// An alternative is to remove the edition when we construct the `Name`. Both look cheap enough.
+
+impl std::hash::Hash for SyntaxContextId {
+    #[inline]
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.remove_root_edition().0.hash(state);
+    }
+}
+
+impl Ord for SyntaxContextId {
+    #[inline]
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.remove_root_edition().0.cmp(&other.remove_root_edition().0)
+    }
+}
+
+impl PartialOrd for SyntaxContextId {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for SyntaxContextId {
+    #[inline]
+    fn eq(&self, other: &Self) -> bool {
+        self.remove_root_edition().0 == other.remove_root_edition().0
+    }
+}
```

Comment:
Doesn't this break interning for salsa? The editions we intern are load bearing, we should do these changes in `Name`'s implementations instead

Reply:
Ah I haven't thought about that, this doesn't seem to break salsa but maybe it does, anyway ignoring them on `Name` was something I thought about as an alternative.